runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/WeightedBoundedQueue.java

```diff
@@ -78,24 +69,26 @@ public void put(V value) {
   public @Nullable V poll(long timeout, TimeUnit unit) throws InterruptedException {
     V result = queue.poll(timeout, unit);
     if (result != null) {
-      limit.release(weigher.apply(result));
+      weigher.release(result);
     }
     return result;
   }
 
   /** Returns and removes the next value, or blocks until one is available. */
   public @Nullable V take() throws InterruptedException {
     V result = queue.take();
-    limit.release(weigher.apply(result));
+    weigher.release(result);
     return result;
   }
 
   /** Returns the current weight of the queue. */
```

Comment:
can this be removed and just look at the weighted semaphore?  It is confusing since it doesn't count just bytes in this queue.

Reply:
done